lib/RGBot.js

```diff
@@ -583,196 +1098,360 @@ const RGBot = class {
+            } else {
+                this.#log(`Cannot dig block ${this.getEntityName(block)}, no capable tool`)
             }
-            this.#bot.on('path_reset', checkForInfiniteDig);
-            this.#log(`Digging ${this.getEntityName(block)}`);
-            await this.#bot.dig(block);
-            this.#bot.off('path_reset', checkForInfiniteDig);
-            result = true;
         }
-        return result;
+        return false
     }
 
     /**
      * Locate and dig the closest Block of a given type within a maximum distance from the Bot.
-     * This method will equip the most appropriate tool in the Bot's inventory for this Block type.
+     * This method will equip the most appropriate tool in the bot inventory for this Block type.
      * @param {string} blockType The name of the Block to find and dig
-     * @param {object} options={} Optional parameters
+     * @param {object} [options={}] Optional parameters
      * @param {boolean} [options.partialMatch=false] Find blocks whose name / displayName contains blockType. (Ex. 'log' may find any of 'spruce_log', 'oak_log', etc.)
      * @param {boolean} [options.onlyFindTopBlocks=false] Will not attempt to dig any Blocks that are beneath another Block
      * @param {number} [options.maxDistance=50] Find any Blocks matching the search criteria up to and including this distance from the Bot
      * @param {boolean} [options.skipCollection=false] If true, the Bot will not explicitly attempt to collect drops from the broken Block. This allows the player to control which drops are collected and which ones are ignored
      * @param {boolean} [options.skipClosest=false] Will attempt to locate the next-closest Block. This can be used to skip the closest Block when the Bot encounters an issue collecting it
      * @returns {Promise<boolean>} true if a Block was found and dug successfully or false if a Block was not found or if digging was interrupted
+     *
+     * @deprecated Since 1.2.0, Collection should be a separate bot action/choice from digging. please use the example for new code
+     * @example Deprecation
+     *         let candidateBlock = bot.findBlocks({blockNames: ['name1',...], maxDistance: maxDistance}, blockPointValueFunction, digCostFunction).shift()?.result
+     *         if (candidateBlock) {
+     *             if (await bot.approachAndDigBlock(candidateBlock)) {
+     *                 // you can collect the item here, but ideally you would let the collection decision be part of your main loop next pass
+     *                 // optionally wait a few ticks for the item to spawn, but that is wasting time that could be used making another action/decision
+     *                 const theItem = this.findItemsOnGroundv2({itemNames: ['name1',...], maxDistance: 5}).shift()?.result
+     *                 if (theItem) {
+     *                     return await this.collectItemOnGround(theItem)
+     *                 }
+     *             }
+     *         }
```

Comment:
Similar to my other comments about multiple functions for the same functionality, I am confused as to why this would be deprecated. As a beginning user who wants to just try and dig my first block, this now seems like a strictly worst user experience (for that case). More specifically, this change would add more lines to our starter examples.

As mentioned in my earlier comment, maybe we can rename this to `findAndDigAndCollectBlock(blockType, options)`, and then have the new findAndDigBlock only find and dig? Then we get the best of both worlds.

Reply:
I think that is fine if they get destroyed - I think that gives them motivation to explore and try other examples :) 

But yes I think that would be a good compromise, we have both options, with slightly different names, for both the beginner and simple use case vs the advanced use cases. I think both are extremely important, and equally important.